tests/DSL.Tests.fs

```diff
@@ -105,6 +105,18 @@ let ``basic checks Zero AssertShape``() =
 //    |> DT.toScalar
 //    |> shouldEqual "wcevwo12" (2.0 * 3.0 + 4.0)
 
+(*
+[<Test>]
+let ``basic checks scalar addition, multiplication, subtraction``() = 
+    let f (x: DT<_>) = 1 * x * x * 1 / 1.0 + 4 * x + 5 - 1 + 1
+    let df x = DT.diff f x
```

Comment:
@moloneymb This test fails - is this because AddGradients is not yet adjusted?

```
Starting test execution, please wait...
Failed   basic checks scalar addition, multiplication, subtraction
Error Message:
 System.AccessViolationException : Attempted to read or write protected memory. This is often an indication that other memory is corrupt.
Stack Trace:
   at TensorFlow.FSharp.DSL.Utils.TF_AddGradients(IntPtr graph, IntPtr* ys, Int32 ny, IntPtr* xs, Int32 nx, IntPtr* dx, IntPtr status, IntPtr* dy)
   at TensorFlow.FSharp.DSL.Utils.gen_ops.add_gradients(Tensor[] y, Tensor[] x, FSharpOption`1 dy) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 143
   at <StartupCode$TensorFlow-FSharp>.$FM.dynodes@1350.Invoke(Unit unitVar0) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 1354
   at TensorFlow.FSharp.DSL.ShapeHelpers.memoize[a,b](Dictionary`2 dict, a key, FSharpFunc`2 f) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 617
   at <StartupCode$TensorFlow-FSharp>.$FM.AddGradients@1348-1.Invoke(TFCtxt ctxt) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 1349
   at TensorFlow.FSharp.DSL.DT.ProducesCorrectShape(FSharpFunc`2 f, TFCtxt ctxt, Boolean _canProduceSmallerShape) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 643
   at TensorFlow.FSharp.DSL.DTModule.gradients@1689-1.Invoke(Tuple`2 tupledArg) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 1689
   at TensorFlow.FSharp.DSL.DT.MakeNode(TFCtxt ctxt, Boolean canProduceSmallerShape) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 658
   at TensorFlow.FSharp.DSL.DT.MakeNodeOfCorrectShape(TFCtxt ctxt) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 661
   at <StartupCode$TensorFlow-FSharp>.$FM.outputNodes@700.Invoke(DT value) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 700
   at TensorFlow.FSharp.DSL.DT.PreprocessCore(FSharpFunc`2 code, Tuple`2[] inputShapes, FSharpOption`1 weights) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 700
   at TensorFlow.FSharp.DSL.DT.RunNDArrays(DT[] values, FSharpOption`1 weights) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 718
   at TensorFlow.FSharp.DSL.DT.RunNDArray(DT value, FSharpOption`1 weights) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 724
   at TensorFlow.FSharp.DSL.DT`1.Eval(DT`1 value, FSharpOption`1 weights) in C:\GitHub\dsyme\fsharp-ai-tools\src\TensorFlow.FSharp\FM.fs:line 1550
   at FM.Tests.basic checks scalar addition, multiplication, subtraction() in C:\GitHub\dsyme\fsharp-ai-tools\tests\DSL.Tests.fs:line 113
Standard Output Messages:
 tensor is not allocated.
 tensor is not allocated.
```

Reply:
I haven't gotten to the bottom of this yet. The TF_Add_Gradient shouldn't be used directly. It was done as a work around as we didn't have the function to get the gradients at the time. The tf.add_gradient should be replaced by tf.gradients from TF.Net. 

When I tried it the test ran but gave the wrong results. tf.gradients appears to be working normally when I test it on it's own so I'm not sure what is causing this.